File: app/src/main/java/fr/zankia/stock/model/Category.java
package fr.zankia.stock.model;

import java.util.ArrayList;
import java.util.List;

public class Category {
    private String name;
    private List<Product> products;

    public Category(String name) {
        this(name, new ArrayList<Product>());
    }

    public Category(String name, ArrayList<Product> products) {
        this.name = name;
        this.products = products;
    }

    public List<Product> getProducts() {
        return products;
    }

    public void setProducts(List<Product> products) {
        this.products = products;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Product getProduct(String name) {
        for(Product p : products) {
            if(p.getName().equals(name)) {
                return p;
            }
        }
        return null;
    }

    public void addProduct(String name) {
        products.add(new Product(name));
    }

    public void removeProduct(String name) {
        products.remove(getProduct(name));
    }

    public void addProduct(Product product) {
        products.add(product);
    }
}

File: app/src/main/java/fr/zankia/stock/model/Category.kt
package fr.zankia.stock.model

import java.util.ArrayList

data class Category(
    var name: String = "",
    val products: ArrayList<Product> = ArrayList()
) {

    fun getProduct(name: String): Product {
        for (p in products) {
            if (p.name == name) {
                return p
            }
        }
        return Product(name)
    }

    fun addProduct(name: String) = products.add(Product(name))


    fun removeProduct(name: String) = products.remove(getProduct(name))
}
